src/mozanalysis/sizing.py

```diff
@@ -0,0 +1,383 @@
+                new_or_resurrected_v3
+            ]
+        )
+
+    Args:
+        experiment_name (str): A simple name of this analysis, which is used
+            to create table names when saving results in BigQuery.
+        start_date (str): e.g. '2019-01-01'. First date for historical data to be
+            retrieved.
+        num_dates_enrollment (int, optional): Only include this many dates
+            of dummy enrollments, where clients that satisfy the target
+            criteria are added to the analysis's dataset.
+            If ``None`` then use the maximum number of dates
+            as determined by the metric's analysis window and
+            ``last_date_full_data``. Typically ``7n+1``, e.g. ``8``. The
+            factor '7' removes weekly seasonality, and the ``+1`` accounts
+            for the fact that enrollment typically starts a few hours
+            before UTC midnight.
+        analysis_length (int, optional): Number of days to include for analysis
+        targets_sql (str, optional): The SQL query that was executed to create
+            the targets
+            table; only defined after get_single_window_data method is called.
+        metrics_sql (str, optional): The SQL query that was executed to create the final
+            results table; only defined after get_single_wondow_data method is
+            called.
+
+
+    Attributes:
+        experiment_name (str): Name of the study, used in naming tables
+            related to this analysis.
+        start_date (str): e.g. '2019-01-01'. First date on which enrollment
+            events were received.
+        num_dates_enrollment (int, optional): Only include this many days
+            of enrollments. If ``None`` then use the maximum number of days
+            as determined by the metric's analysis window and
+            ``last_date_full_data``. Typically ``7n+1``, e.g. ``8``. The
+            factor '7' removes weekly seasonality, and the ``+1`` accounts
+            for the fact that enrollment typically starts a few hours
+            before UTC midnight.
+        analysis_length (int, optional): Number of days to include for analysis
+        targets_sql (str): The SQL query that was executed to create the targets
+            table; only defined after get_single_window_data method is called.
+        metrics_sql (str): The SQL query that was executed to create the final
+            results table; only defined after get_single_wondow_data method is
+            called.
+
+    """
+
+    experiment_name = attr.ib()
+    start_date = attr.ib()
+    num_dates_enrollment = attr.ib(default=None)
+    analysis_length = attr.ib(default=None)
+    targets_sql = attr.ib(default='')
+    metrics_sql = attr.ib(default='')
+
+    def get_single_window_data(
+        self,
+        bq_context,
+        metric_list,
+        last_date_full_data,
+        analysis_start_days,
+        analysis_length_days,
+        target_list=None,
```

Comment:
Given the move to `Segment` objects, should this be `segment_list`?

Reply:
It could, but I kept this naming to note the distinction between how Segment objects are used here, where they define the exact client IDs that are included, versus how they're used in Experiment, where they add a binary column indicating segment inclusion

Reply:
Ahh, makes sense. 👍